File: my-lol-game/src/edu/lehigh/cse/lol/Route.java

package edu.lehigh.cse.lol;

/**
 * A Route specifies a set of points that an entity will move between at a fixed
 * speed.
 */
public class Route {
    /**
     * The X coordinates of the points in the route
     */
    float[] _xIndices;

    /**
     * The Y coordinates of the points in the route
     */
    float[] _yIndices;

    /**
     * The speed at which the entity should move along the route
     */
    float _velocity;

    /**
     * The maximum number of points in this route
     */
    int _size;

    /**
     * The current number of points that have been set
     */
    int _points;

    /**
     * Define a new path, by specifying the number of points in the path. Note
     * that all points in the path will be uninitialized until the "to" method
     * is called on this Route.
     * 
     * @param numberOfPoints number of points in the path
     */
    public Route(int numberOfPoints) {
        // NB: it doesn't make sense to have a route with only one point!
        assert (numberOfPoints > 1);
        _size = numberOfPoints;
        _xIndices = new float[_size];
        _yIndices = new float[_size];
    }

    /**
     * Add a new point to a path by giving (x,y) coordinates for where the
     * center of the object ought to move
     * 
     * @param x X value of the new coordinate
     * @param y Y value of the new coordinate
     */
    public Route to(float x, float y) {
        _xIndices[_points] = x;
        _yIndices[_points] = y;
        _points++;
        return this;
    }
}

File: LOL02/LOL02/LOL02/LOL/Route.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Diagnostics;

namespace LOL
{
    public class Route
    {
        /**
         * The X coordinates of the points in the route
         */
        public float[] mXIndices;

        /**
         * The Y coordinates of the points in the route
         */
        public float[] mYIndices;

        /**
         * The speed at which the entity should move along the route
         */
        public float mVelocity;

        /**
         * The maximum number of points in this route
         */
        public int mSize;

        /**
         * The current number of points that have been set
         */
        public int mPoints;

        /**
         * Define a new path, by specifying the number of points in the path. Note
         * that all points in the path will be uninitialized until the "to" method
         * is called on this Route.
         * 
         * @param numberOfPoints number of points in the path
         */
        public Route(int numberOfPoints)
        {
            // NB: it doesn't make sense to have a route with only one point!
            Debug.Assert(numberOfPoints > 1);
            mSize = numberOfPoints;
            mXIndices = new float[mSize];
            mYIndices = new float[mSize];
        }

        /**
         * Add a new point to a path by giving (x,y) coordinates for where the
         * center of the object ought to move
         * 
         * @param x X value of the new coordinate
         * @param y Y value of the new coordinate
         */
        public Route to(float x, float y)
        {
            mXIndices[mPoints] = x;
            mYIndices[mPoints] = y;
            mPoints++;
            return this;
        }
    }
}
